packages/typography/src/elements/CodeBlock.tsx

```diff
@@ -18,13 +24,41 @@ export interface ICodeBlockProps extends HTMLAttributes<HTMLPreElement> {
   isLight?: boolean;
   /** Displays line numbers */
   isNumbered?: boolean;
+  /** Determines the lines to highlight */
+  highlightLines?: number[];
+  /** Passes props to the code block container */
+  containerProps?: HTMLAttributes<HTMLDivElement>;
 }
 
 /**
  * @extends HTMLAttributes<HTMLPreElement>
  */
 export const CodeBlock = React.forwardRef<HTMLPreElement, ICodeBlockProps>(
-  ({ children, isLight, isNumbered, language, size, ...other }, ref) => {
+  (
+    { children, containerProps, highlightLines, isLight, isNumbered, language, size, ...other },
+    ref
+  ) => {
+    const [containerTabIndex, setContainerTabIndex] = useState(-1);
+    const containerRef = useRef<HTMLDivElement>(null);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Sometimes the eslint `rules-of-hooks` can get confused when not using an immediate inline function with `useCallback`. Since you're returning a reference, `useMemo` might be more applicable here. (we ran into this with the ColorPicker PR)

Reply:
After giving this a closer look, I'm not convinced there's much here that benefits from memoization. Please take a look at the updates and let me know what you think.